File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int ans = -1;
        
        for(int i=1; i<arr.size()-1; i++)
        {
            if(arr[i-1] < arr[i] && arr[i] > arr[i+1])
            {
                ans = i;
                break;
            }
        }
        
        return ans;
        
    }
};

File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        start = 0
        end = len(arr) - 1
        result = -1
        
        while start <=end :
            mid = (start+end)//2
            if arr[mid] > arr[mid + 1]:
                result = mid
                end = mid - 1
            else:
                start = mid + 1
        
        return result
        